packages/now-node/src/index.ts

```diff
@@ -474,27 +478,49 @@ async function doTypeCheck({
   // In order to type-check a single file, a temporary tsconfig
   // file needs to be created that inherits from the base one :(
   // See: https://stackoverflow.com/a/44748041/376773
-  const id = Math.random()
-    .toString(32)
-    .substring(2);
-  const tempConfigName = `.tsconfig-${id}.json`;
+  const absoluteEntrypoint = join(workPath, entrypoint);
   const projectTsConfig = await walkParentDirs({
     base: workPath,
     start: join(workPath, dirname(entrypoint)),
     filename: 'tsconfig.json',
   });
+  const sha = createHash('sha256')
+    .update(absoluteEntrypoint)
+    .update(projectTsConfig || '')
+    .digest('hex');
+  const tempConfigName = `vercel-node-${sha}.json`;
+  const tempConfigPath = join(TMP, tempConfigName);
   const tsconfig = {
     extends: projectTsConfig || undefined,
-    include: [entrypoint],
+    include: [absoluteEntrypoint],
   };
-  await fsp.writeFile(tempConfigName, JSON.stringify(tsconfig));
+
+  console.error({
+    absoluteEntrypoint,
+    sha,
+    tempConfigName,
+    tempConfigPath,
+    tsconfig,
+  });
+
+  try {
+    await fsp.writeFile(tempConfigName, JSON.stringify(tsconfig), {
+      flag: 'wx',
+    });
+  } catch (err) {
+    console.error(err);
+    // Don't throw if the file already exists
+    if (err.code !== 'EEXIST') {
```

Comment:
Seems like the `w` flag alone will have the same behavior. Why use the `x` flag if you're going to catch?

Reply:
We want the write to happen only once, and be an atomic operation. If every HTTP request is going to regenerate the file then there's race conditions for the type checking vs. the time it takes to truncate the file and rewrite it to the FS. The contents will always be identical anyways.

Reply:
I see, I just assumed that try/catch every time would be slower than overwriting the file each time.